cygnus-arcgis/src/main/java/com/telefonica/iot/cygnus/backends/arcgis/restutils/RestFeatureTable.java

```diff
@@ -0,0 +1,590 @@
+        }
+    }
+
+    /**
+     * 
+     * @param url
+     * @param credential
+     * @param tokenGenUrl
+     * @param parentLogger
+     * @throws ArcgisException
+     */
+    public RestFeatureTable(String url, Credential credential, String tokenGenUrl,
+            BaseLoggerInterface parentLogger) throws ArcgisException {
+        this(url, credential, tokenGenUrl);
+        this.parentLogger = parentLogger;
+    }
+
+    /**
+     * Constructor
+     * 
+     * @param url
+     * @param credential
+     * @throws MalformedURLException
+     */
+    // public RestFeatureTable (URL url, Credential credential, URL tokenGenUrl){
+    // super(tokenGenUrl, credential, url.toString());
+    // this.serviceUrl = url;
+    // }
+
+    /**
+     * @return the tableAttributes
+     */
+    public Map<String, Field> getTableAttributes() {
+        return tableAttributes;
+    }
+
+    /**
+     * @param tableAttributes
+     *            the tableAttributes to set
+     */
+    public void setTableAttributes(Map<String, Field> tableAttributes) {
+        this.tableAttributes = tableAttributes;
+    }
+
+    /**
+     * @return the uniqueAttributes
+     */
+    public Map<String, Boolean> getUniqueAttributes() {
+        return uniqueAttributes;
+    }
+
+    /**
+     * @param uniqueAttributes
+     *            the uniqueAttributes to set
+     */
+    public void setUniqueAttributes(Map<String, Boolean> uniqueAttributes) {
+        this.uniqueAttributes = uniqueAttributes;
+    }
+
+    /**
+     * @return the uniqueIdField
+     */
+    public String getUniqueIdField() {
+        return uniqueIdField;
+    }
+
+    /**
+     * @return the uniqueIdField
+     */
+    public boolean hasUniqueIdField() {
+        return this.tableAttributes.containsKey(uniqueIdField);
+    }
+
+    /**
+     * 
+     * @param whereClause
+     * @return
+     * @throws ArcgisException
+     */
+    public List<Feature> getFeatureList(String whereClause) throws ArcgisException {
+        List<Feature> resultList = new ArrayList<Feature>();
+        ResultPage<Feature> page;
+        int pageOffset = 0;
+
+        do {
+            page = getFeatureList(whereClause, pageOffset, getCredential().getToken());
+            pageOffset += page.getItemsSize();
+            resultList.addAll(page.asItemsList());
+        } while (page.hasNext());
+
+        return resultList;
+    }
+
+    /**
+     * 
+     * @param whereClause
+     * @param token
+     * @return
+     * @throws ArcgisException
+     */
+    public ResultPage<Feature> getFeatureList(String whereClause, int pageOffset, String token)
+            throws ArcgisException {
+
+        logDebug("getFeatureList - Connecting Feature table: " + serviceUrl);
+
+        String responseJSON = null;
+
+        Map<String, String> params = new LinkedHashMap<String, String>();
+        params.put("outSR", "{\"wkid\":4326}");
+        params.put("outFields", "*");
```

Comment:
It would be better to declare this Strings as constants.

Reply:
Fixed https://github.com/Stdri/fiware-cygnus/commit/042bbabcf92945d2fd086d5bf08b1f3370182e84